### DISCUSSION TITLE
Password expiration update in freeipa

### DISCUSSION THREAD

**Discussion by cybertschunk** [keycloak/keycloak#21449] [Q&A] [answered]
Hello dear all,
I'm strugling to integrate keycloak with our FreeIPA installation. We would like to use the password update as well as the password reset feature.

FreeIPA uses the 'krbPasswordExpiration' field of an user object to store the information when the password expire. Unfortunately, Keycloak does not update this field when a password is changed. This way the passwords expire the very day they are set. We configured a password policy that should give the password a lifespan of one year.

I also couldn't find a mapper for this in the LDAP binding. Do you have any idea how to deal with this problem? Is there a capable workaround?

Have a nice day!
David

**Comment by nyetsche** [comment]
Sadly, after tearing my hair out about this, the only workaround is to build your own plugin that sends the updated attribute yourself. FreeIPA is insistent that any password change that isn't done by the actual user (through an `ipa` command) gets a password that expires immediately.

I hacked together code that could do this and it did "work", but I don't do much Java development, even less Keycloak development, and we eventually moved away from Keycloak/FreeIPA for this and other reasons.

But if anyone wants to try again, here's what I came up with - https://gist.github.com/nyetsche/f7aaa5fde4a19e79440e9a05161a59f4

**Comment by gasinvein** [comment, marked as answer]
If the problem is still relevant for you - as a workaround, you can add an exemption on the FreeIPA server for the user Keycloak is binding to LDAP as, so that the password isn't imminently expire after being reset by Keycloak.
See https://github.com/keycloak/keycloak/issues/26945#issuecomment-2363272250
